presto-router/src/test/java/com/facebook/presto/router/TestCustomSchedulerManager.java

```diff
@@ -0,0 +1,233 @@
+    private HttpServerInfo httpServerInfo;
+    private File configFile;
+    private CustomSchedulerManager schedulerManager;
+    private static final Random RANDOM = new Random();
+    private static List<URI> serverURIs;
+
+    @BeforeClass
+    public void setup()
+            throws Exception
+    {
+        Logging.initialize();
+
+        // Set up servers
+        ImmutableList.Builder<TestingPrestoServer> builder = ImmutableList.builder();
+        for (int i = 0; i < NUM_CLUSTERS; ++i) {
+            builder.add(createPrestoServer());
+        }
+        prestoServers = builder.build();
+
+        configFile = getRouterConfigFile(prestoServers);
+
+        Path tempPluginSchedulerConfigFile = Files.createTempFile("router-scheduler-mock", ".properties");
+        String schedulerName = "router-scheduler.name=mock";
+        Files.write(tempPluginSchedulerConfigFile, schedulerName.getBytes());
+
+        SchedulerFactory mockSchedulerFactory = new MockSchedulerFactory();
+        CustomSchedulerManager customSchedulerManager = new CustomSchedulerManager(ImmutableList.of(mockSchedulerFactory), tempPluginSchedulerConfigFile.toFile());
+
+        Bootstrap app = new Bootstrap(
+                new TestingNodeModule("test"),
+                new TestingHttpServerModule(),
+                new JsonModule(),
+                new JaxrsModule(true),
+                new RouterModule(Optional.of(customSchedulerManager)));
+
+        Injector injector = app.doNotInitializeLogging()
+                .setRequiredConfigurationProperty("router.config-file", configFile.getAbsolutePath())
+                .quiet().initialize();
+
+        httpServerInfo = injector.getInstance(HttpServerInfo.class);
+        schedulerManager = injector.getInstance(CustomSchedulerManager.class);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDownServer()
+            throws Exception
+    {
+        for (TestingPrestoServer prestoServer : prestoServers) {
+            prestoServer.close();
+        }
+    }
+
+    static class MockSchedulerFactory
+            implements SchedulerFactory
+    {
+        public static final String MOCK = "mock";
+
+        @Override
+        public String getName()
+        {
+            return MOCK;
+        }
+
+        @Override
+        public Scheduler create(Map<String, String> config)
+        {
+            return new MockScheduler();
+        }
+    }
+
+    public static class MockScheduler
+            implements Scheduler
+    {
+        private List<URI> candidates;
+        private int requestsMade;
+
+        @Override
+        public Optional<URI> getDestination(String user)
+        {
+            return Optional.empty();
+        }
+
+        @Override
+        public Optional<URI> getDestination(String user, String query)
+        {
+            ++requestsMade;
+            return Optional.of(serverURIs.get(RANDOM.nextInt(serverURIs.size())));
```

Comment:
Since we're setting candidates, lets use one of them instead ? Also, lets pick a predefined one, say `candidates.get(0)`, no need for random here